internal/api/v1/auth.go

```diff
@@ -17,66 +17,96 @@ import (
 	"strings"
 
 	"github.com/epinio/epinio/internal/api/v1/response"
-	"github.com/epinio/epinio/internal/auth"
 	"github.com/epinio/epinio/internal/cli/server/requestctx"
 	apierrors "github.com/epinio/epinio/pkg/api/core/v1/errors"
 	"github.com/gin-gonic/gin"
 	"github.com/go-logr/logr"
 )
 
-func AuthorizationMiddleware(c *gin.Context) {
+func NamespaceAuthorizationMiddleware(c *gin.Context) {
+	user := requestctx.User(c.Request.Context())
+	authorization(c, "namespace", user.Namespaces)
+}
+
+func GitconfigAuthorizationMiddleware(c *gin.Context) {
+	user := requestctx.User(c.Request.Context())
+	authorization(c, "gitconfig", user.Gitconfigs)
+}
+
+func authorization(c *gin.Context, label string, allowed []string) {
 	logger := requestctx.Logger(c.Request.Context()).WithName("AuthorizationMiddleware")
 	user := requestctx.User(c.Request.Context())
 
 	method := c.Request.Method
 	path := c.Request.URL.Path
-	namespace := c.Param("namespace")
 
 	logger.Info(fmt.Sprintf("authorization request from user [%s] with role [%s] for [%s - %s]", user.Username, user.Role, method, path))
 
-	var authorized bool
 	switch user.Role {
 	case "admin":
-		authorized = authorizeAdmin(logger)
+		logger.V(1).WithName("authorizeAdmin").Info("user [admin] is authorized")
+		return
+
 	case "user":
-		authorized = authorizeUser(logger, user, path, namespace)
+		if !unrestrictedPath(logger, path) {
+			response.Error(c, apierrors.NewAPIError("user unauthorized, path restricted",
+				http.StatusForbidden))
+			c.Abort()
+		}
```

Comment:
The negation of a negative verb is a bit redundant/confusing. Would you agree on change this to:

```go
if restrictedPath(logger, path) {
    // abort
}
```
or
```go
if !allowedPath(logger, path) {
    // abort
}
```

Also, I think we should return after the `c.Abort()`, since the middleware failed.

Reply:
Oops and yes.

Reply:
Done with commit 87f505aebd12d1d18aa5eeab13042058f94b9b1a